Improve create_distributed_hypertable (#533)
timescale/docs.timescale.com-content: api.md

| `partitioning_func` | The function to use for calculating a value's partition.|
| `associated_schema_name` | Name of the schema for internal hypertable tables. Default is "_timescaledb_internal". |
| `associated_table_prefix` | Prefix for internal hypertable chunk names. Default is "_hyper". |

> | `migrate_data` | Set to `true` to migrate any existing `main_table` data to chunks in the new hypertable. A non-empty table will generate an error without this option. Note that, for large tables, the migration might take a long time. Defaults to FALSE. |

Reviewer: ```suggestion
| `migrate_data` | Set to `true` to migrate any existing data from the `relation` table to chunks in the new hypertable. A non-empty table will generate an error without this option. Large tables may take significant time to migrate. Defaults to FALSE. |
```

|Name|Description|
|---|---|
| `partitioning_column` | Name of an additional column to partition by. |
| `number_partitions` | Number of hash partitions to use for `partitioning_column`. Must be > 0. Default is the number of `data_nodes`. |
| `associated_schema_name` | Name of the schema for internal hypertable tables. Default is "_timescaledb_internal". |
| `associated_table_prefix` | Prefix for internal hypertable chunk names. Default is "_hyper". |
| `chunk_time_interval` | Interval in event time that each chunk covers. Must be > 0. As of TimescaleDB v0.11.0, default is 7 days, unless adaptive chunking (DEPRECATED)  is enabled, in which case the interval starts at 1 day. For previous versions, default is 1 month. |
| `create_default_indexes` | Boolean whether to create default indexes on time/partitioning columns. Default is TRUE. |
| `if_not_exists` | Boolean whether to print warning if table already converted to hypertable or raise exception. Default is FALSE. |
| `partitioning_func` | The function to use for calculating a value's partition.|

> | `migrate_data` | Set to `true` to migrate any existing `relation` data to chunks in the new hypertable. A non-empty table will generate an error without this option. Note that, for large tables, the migration might take a long time. Defaults to FALSE. |

Reviewer 2: Not sure if this is intentional. You are using TRUE above, but `true` here.
```suggestion
| `migrate_data` | Set to TRUE to migrate any existing `relation` data to chunks in the new hypertable. A non-empty table will generate an error without this option. Note that, for large tables, the migration might take a long time. Defaults to FALSE. |
```

Author: @mkindahl It is not me, it was there. I homogenised only places where it wasn't code.

Author: We need decision about consistent way of using Boolean constants, since it is all possible cases in the file.

Reviewer: Pending internal poll, I'll post the decision =)

Author: Homogenise a little bit, based on the current discussion. However, I suggest to do such text refactoring separately.

> | `migrate_data` | Set to `true` to migrate any existing `relation` data to chunks in the new hypertable. A non-empty table will generate an error without this option. Note that, for large tables, the migration might take a long time. Defaults to FALSE. |

Reviewer: ```suggestion
| `migrate_data` | Set to `true` to migrate any existing data from the `relation` table to chunks in the new hypertable. A non-empty table will generate an error without this option. Large tables may take significant time to migrate. Defaults to FALSE. |
```

| `time_partitioning_func` | Function to convert incompatible primary time column values to compatible ones. The function must be `IMMUTABLE`. |

> | `replication_factor` | This will determine the number of data nodes each incoming write is written to.  The default value is 1 and this is required to be 1 or greater. Read [the best pracitce](#create_distributed_hypertable-best-practices) below if you consider non-default values. |

Reviewer 2: ```suggestion
| `replication_factor` | This will determine the number of data nodes each incoming write is written to.  The default value is 1 and this is required to be 1 or greater. Read [the best practice](#create_distributed_hypertable-best-practices) below if you consider non-default values. |
```

> | `replication_factor` | This will determine the number of data nodes each incoming write is written to.  The default value is 1 and this is required to be 1 or greater. Read [the best practice](#create_distributed_hypertable-best-practices) below if you consider non-default values. |

Reviewer: ```suggestion
| `replication_factor` | The number of data nodes to which is each inserted row is written (and implicitly, the replication factor of a chunk).  Must be >= 1; defaults to 1.  Read [the best practices](#create_distributed_hypertable-best-practices) before changing the default. |
```

> | `data_nodes` | This is the set of data nodes that will be used for this table.  If this is not present then all data nodes known by this access node instance will be used to distribute the hypertable. |

Reviewer: ```suggestion
| `data_nodes` | The set of data nodes used for the distributed hypertable.  If not present, defaults to all data nodes known by the access node (the node on which the distributed hypertable is created). |
```

should be the same as the non-distributed case, as all of the incoming
data will be handled by a single node.

> **Replication factor:** `relication_factor` defines to how many data nodes
> newly created chunk will be replicated. So if a data node is removed, no data
> will be lost, but the chunks, which were present on the removed data node, will
> be less replicated, underreplicated. Currently it is not possible to get back the 
> replication of the underreplicated chunks. While newly created chunks will be created
> with the required number of replicas. To avoid such inconsistency we don't recommend
> to use `replication_factor` bigger than 1.

Reviewer: ```suggestion
**Replication factor:**  The hypertable's `replication_factor` defines to how
many data nodes a newly created chunk will be replicated.  That is, a chunk
with a `replication_factor` of three will exist on three separate data nodes,
and rows written to that chunk will be inserted (as part of a two-phase
commit protocol) to all three chunk copies.  For chunks replicated more
than once, if a data node fails or is removed, no data will be lost, and writes
can continue to succeed on the remaining chunk copies.  However, the chunks
present on the lost data node will now be under-replicated than the specified
replication factor.  Currently, it is not possible to restore under-replicated
chunks, although this limitation will be removed in a future release. To avoid
such inconsistency, we do not yet recommend using `replication_factor` > 1,
and instead rely on physical replication of each data node.
```
